databuilder/databuilder/extractor/snowflake_usage_extractor.py

```diff
@@ -0,0 +1,106 @@
+"""
+The Snowflake Usage Extractor allows you to populate the "Frequent Users" and "Popular Tables" features within
+Amundsen with the help of the TableColumnUsage class. The way you calculate popularity for Snowflake is custom
+and an example of how to calculate table popularity can be found in the sample_snowflake_table_usage.scala script.
+"""
+
+import logging
+from typing import (
+    Dict, Iterator, Union,
+)
+
+from pyhocon import ConfigFactory, ConfigTree
+from unidecode import unidecode
+
+from databuilder import Scoped
+from databuilder.extractor import sql_alchemy_extractor
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_column_usage import ColumnReader, TableColumnUsage
+
+LOGGER = logging.getLogger(__name__)
+
+class SnowflakeUsageExtractor(Extractor):
+    # SELECT statement from snowflake table that contains usage data
+    SQL_STATEMENT = """
+        SELECT
+            database,
+            schema,
+            name,
+            user_email,
+            read_count
+        FROM
+            {database}.{schema}.{table}
```

Comment:
Given that this table will not be available by default in snowflake and that some pre-processing needs to occur to create this table, should this executor include the curation of this table _and then_ the extraction of this metadata?

Reply:
That's a great suggestion. I do think that wherever you're writing the data to though should include the table creation. You'd still need info about the database and schema, and then the table name would be variable. So pre-processing would still need to happen if you want to place this in a specific separate db/schema. Does that sound reasonable?

Reply:
I think the point here also would be - what makes this extractor `Snowflake` specific? We have a very similar table in hive only with parsed Apache Ranger Logs. The columns are much less identical (the names vary slightly). For me Snowflake specific part is one step before (where you create this table using scala parser) - maybe we could treat this extractor in a more generic fashion?

Reply:
+1 to this. The structure of the table you've defined seems quite reasonable and something that I think is a good idea to standardize on:
```
SELECT
        database,
        schema,
        name,
        user_email,
        read_count
    FROM
{database}.{schema}.{table}
```
Can this extractor take a reference to the database driver (e.g. pg, snowflake, etc.) so we can reuse it?

Reply:
If we standardize then this should be `user_id` as different companies use different attributes for id.

Reply:
Ya I kept the table the same as the one that `TableColumnUsage` reads in: https://github.com/amundsen-io/amundsen/blob/main/databuilder/databuilder/models/table_column_usage.py#L25-L31, would we want to deviate from this?

Reply:
I also didn't think about making this a generic extractor because I'm not familiar with that many other database drivers. Would the only information you'd need is the type of database? (like just making the `database` variable a variable instead of me hardcoding it as `snowflake`?)

Reply:
https://github.com/amundsen-io/amundsen/pull/1192#discussion_r645361211